File: FinanceAssistant/Core/Models/Transaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace FinanceAssistant.Core.Models
{
    public class Transaction
    {
        public int Id { get; set; }

        [Required]
        public int TypeId { get; set; } // foreign key
        public TransactionType Type { get; set; }

        [Required]
        public int CategoryId { get; set; } // foreign key
        public TransactionCategory Category { get; set; }

        public string Description { get; set; }

        [Required]
        public int Amount { get; set; }

        [Required]
        public string Currency { get; set; }

        [Required]
        public DateTime Date { get; set; }
    }
}

File: FinanceAssistant/ClientApp/app/models/transaction.ts
export interface Type {
    id: number;
    name: string;
}

export interface Category {
    id: number;
    name: string;
    type: Type;
}

export interface Transaction {
    id: number;
    category: Category;
    description: string;
    amount: number;
    currency: string;
    date: Date;
}

export interface SaveTransaction {
    id: number;
    typeId: number;
    categoryId: number;
    description: string;
    amount: number;
    currency: string;
    date: Date;
}